R/content-plots.R

```diff
@@ -4,23 +4,29 @@
 #' reference, this function will return a vector of the same length as the
 #' granges object, with percent of Cs and Gs.
 #' 
-#' @param query  A GenomicRanges or GenomicRangesList object with query regions
-#' @param ref BSgenome object
+#' @param query  A GenomicRanges or GenomicRangesList object with query regions.
+#' @param ref Reference genome BSgenome object.
+#' @return A numeric vector of list of vectors with the GC percentage of the query regions.
 #' @export
 #' @examples
-#' \dontrun{
-#' bsg = loadBSgenome('hg19')
-#' gcvec = calcGCContent(query, bsg)
-#' }
+#'bsg = loadBSgenome('hg19')
+#'gcvec = calcGCContent(query, bsg)
+#' 
 calcGCContent = function(query, ref) {
 	# if (!requireNamespace(ref, quietly=TRUE)) {
 	# 	message(ref, " package is not installed.")
 	# }
-	if (is(query, "GRangesList")) {
-		# Recurse over each GRanges object
-		x = lapply(query, calcGCContent, ref)
-		return(x)
-	}	
+	if (!(is(query, "GRanges") || is(query, "GRangesList" ))) {
```

Comment:
the input class validation is duplicated in couple of functions. Let's implement a generic function that does that. It could be maintained in a single place this way